src/backend/postmaster/pgstat.c

```
@@ -2557,9 +2557,22 @@ pgstat_fetch_resgroup_queue_timestamp(void)
  * NB: this *must* be able to survive being called before MyBEEntry has been
  * initialized.
  * ----------
+ *
+ * In GPDB, this interface is modified to accept a reason for waiting while in
+ * upstream, it just accepts a boolean value.  The interface is renamed in GPDB
+ * to "gpstat_report_waiting(char waiting)" in order to catch future uses of
+ * the interface when merged from upstream.
  */
+#ifdef USE_ASSERT_CHECKING
```

why not delete or `#if 0` instead and fail faster ?